### ISSUE TITLE
Feature request: ability to parse an html fragment with includeNodeLocations

### ISSUE THREAD

**Issue by rictic** [jsdom/jsdom#3238]
When using jsdom to work with html fragments (e.g. to process Lit templates in a build pipeline), it would be very useful to be able to get source location information (e.g. to report diagnostics).

The `fragment()` function currently doesn't take any options.

Currently the best work around we have is to parse the fragment as a document and postprocess the HTML (e.g. by removing synthetic nodes inserted by the parser), but there's a lot of edge cases (e.g. around table elements) where that becomes very difficult.

**Comment by domenic** [comment]
We can't really do this because the entire point of the fragment() function is to reuse the same document, and that document has no options.

**Comment by domenic** [comment]
You should just be able to do it yourself, fragment is like 6 lines: https://github.com/jsdom/jsdom/blob/ba8e5806d1a23960a03c3dd99af98fd0916b4bc9/lib/api.js#L102-L108

No need for messy node removal.

**Comment by rictic** [comment]
Hm, so something like:

```js
const dom = new JSDOM(`<template>\n${fragmentText}</template>`, {includeNodeLocations: true});
```

And then fixup all the node locations by subtracting 1 line?

**Comment by domenic** [comment]
Yeah, although that won't be quite right with respect to stuff like `<b>something</b></template>something else`, probably.

I guess node locations aren't included if you just use `includeNodeLocations` plus `template.innerHTML`?

**Comment by rictic** [comment]
You mean like this?

```js
const jsdom = require("jsdom");
const dom = new jsdom.JSDOM('<template></template>', { includeNodeLocations: true });
const template = dom.window.document.querySelector('template');
template.innerHTML = '<div></div>';
// logs undefined
console.log(dom.nodeLocation(template.content.querySelector('div')));
```

**Comment by domenic** [comment]
Yeah. Sad that doesn't work... although I guess it kind of makes sense, as otherwise you could have multiple things at the same location.
